### ISSUE TITLE
huggingface-imported ALBERT model doesn't train at all

### ISSUE THREAD

**Issue by aarzchan** [pytorch/fairseq#3013]
#### Issue
I'm trying to use huggingface's ALBERT encoder in fairseq, following the [HuggingFaceGPT2LanguageModel example](https://github.com/pytorch/fairseq/blob/master/fairseq/models/huggingface/hf_gpt2.py). However, although ALBERT seems to have been imported into fairseq successfully, it does not train at all, yielding a constant train loss of 1.61. 

For reference, I'm doing multiple-choice QA on the CommonsenseQA dataset. The model is supposed to output a different score for each choice, but it always outputs the same score for all choices. Meanwhile, fairseq's built-in RoBERTa encoder runs fine on the same fairseq dataset (while using RoBERTa tokenization).

#### Code

Here is my ALBERT wrapper class. I just put `pass` in `base_v2_architecture` and `large_v2_architecture`, since all the necessary architecture args are already loaded in the `from_pretrained` call.
```
arch_dict = {
    'qa__albert_base_v2': 'albert-base-v2',
    'qa__albert_large_v2': 'albert-large-v2'
}

@register_model("albert")
class AlbertModel(FairseqEncoderModel):
    def __init__(self, encoder):
        super().__init__(encoder)

    @classmethod
    def build_model(cls, args, task):
        """Build a new model instance."""
        base_v2_architecture(args)
        return cls(AlbertEncoder(args, task))

    def forward(self, src_tokens, **kwargs):
        return self.encoder(src_tokens, **kwargs)


class AlbertEncoder(FairseqEncoder):
    def __init__(self, args, task):
        try:
            from transformers import AlbertModel
        except ImportError:
            raise ImportError(
                "\n\nPlease install huggingface/transformers with:"
                "\n\n  pip install transformers"
            )

        super().__init__(task.dictionary)
        self.model = AlbertModel.from_pretrained(arch_dict[args.arch])
        self.output_layer = encoder_head_dict[args.text_args['encoder_head']](args)
        self.pad_idx = self.model.config.pad_token_id

    def forward(
        self,
        src_tokens,
        features_only=False,
        return_all_hiddens=False,
        masked_tokens=None,
        **unused
    ):
        x, extra = self.extract_features(
            src_tokens, return_all_hiddens=return_all_hiddens
        )
        if not features_only:
            x = self.output_layer(x, masked_tokens=masked_tokens)
        return x, extra

    def extract_features(self, src_tokens, return_all_hiddens=False, **kwargs):
        features, _ = self.model(
            input_ids=src_tokens,
            attention_mask=src_tokens.eq(self.pad_idx),
        )
        return features, {}

def max_positions(self):
    """Maximum output length supported by the encoder."""
    return self.model.config.max_position_embeddings

@register_model_architecture("albert", "albert_base_v2")
def base_v2_architecture(args):
    pass

@register_model_architecture("albert", "albert_large_v2")
def large_v2_architecture(args):
    pass
```

   

#### What I've tried

- Checked that `requires_grad=True` for all model parameters.
- Checked the model parameters are not changing after each "update".
- Checked that encoder inputs look normal. Each choice has a different ALBERT-tokenized input_ids tensor.
- Checked the encoder's output embeddings. Strangely, every single output embedding (not just BOS embedding) is the same, across both choices and instances.
- Checked the pretrained ALBERT weights are being loaded properly. The pretrained weights loaded in fairseq match those loaded in a python shell, using `model = AlbertModel.from_pretrained('albert-base-v2')`.
- Tried changing the learning rate and batch size.
- Tried using both albert-base-v2 and albert-large-v2 architectures.
- Tried importing RoBERTa-Base and BERT-Base from huggingface, and a very similar thing happened. Instead of the train loss staying constant at 1.61, the train loss now oscillates around this value.
- Tried upgrading huggingface transformers to version 4.0.0 (latest version).
- Tried manually loading ALBERT weights and config.

#### Environment

 - fairseq Version: master
 - PyTorch Version: 1.7.0
 - huggingface transformers Version: 3.5.1
 - OS: Linux
 - How you installed fairseq: source
 - Build command you used (if compiling from source): `pip install --editable ./`
 - Python version: 3.8.5
 - CUDA/cuDNN version: 10.1
 - GPU models and configuration: GeForce GTX 1080, Quadro RTX 6000, Quadro RTX 8000

**Comment by CheungZeeCn** [comment]
My implementation for Chinese bert, may it helps you.
https://github.com/CheungZeeCn/fairseq/blob/master/Sentence_prediction_Chinese_bert_demo.md

**Comment by aarzchan** [comment]
Thanks for the example, @CheungZeeCn!

I was able to figure out the issue: my attention mask was flipped. In `extract_features`, it should be `attention_mask=src_tokens.ne(self.pad_idx)` instead of `attention_mask=src_tokens.eq(self.pad_idx)`.
